Resources/doc/3-6-param_fetcher_listener.rst

```diff
@@ -0,0 +1,142 @@
+         * resut and is likely to be considered as a bug) you can precise the parameters can't be present
+         * at the same time by doing
+         *
+         * @RequestParam(name="search", requirements="[a-z]+", description="search")
+         * @RequestParam(name="byauthor", requirements="[a-z]+", description="by author", incompatibles={"search"})
+         *
+         * If you want to work with array: ie. ?ids[]=1&ids[]=2&ids[]=1337, use:
+         *
+         * @QueryParam(array=true, name="ids", requirements="\d+", default="1", description="List of ids")
+         * (works with QueryParam and RequestParam)
+         *
+         * It will validate each entries of ids with your requirement, by this way, if an entry is invalid,
+         * this one will be replaced by default value.
+         *
+         * ie: ?ids[]=1337&ids[]=notinteger will return array(1337, 1);
+         * If ids is not defined, array(1) will be given
+         *
+         * Array must have a single depth or it will return default value. It's difficult to validate with
+         * preg_match each deeps of array, if you want to deal with that, you can use a constraint:
+         *
+         * @QueryParam(array=true, name="filters", requirements=@MyComplexConstraint, description="List of complex filters")
+         *
+         * In this example, the ParamFetcher will validate each value of the array with the constraint, returning the
+         * default value if you are in safe mode or throw a BadRequestHttpResponse containing the constraint violation
+         * messages in the message.
+         *
+         * @param ParamFetcher $paramFetcher
+         */
+        public function getArticlesAction(ParamFetcher $paramFetcher)
+        {
+            // ParamFetcher params can be dynamically added during runtime instead of only compile time annotations.
+            $dynamicRequestParam = new RequestParam();
+            $dynamicRequestParam->name = "dynamic_request";
+            $dynamicRequestParam->requirements = "\d+";
+            $paramFetcher->addParam($dynamicRequestParam);
+
+            $dynamicQueryParam = new QueryParam();
+            $dynamicQueryParam->name = "dynamic_query";
+            $dynamicQueryParam->requirements="[a-z]+";
+            $paramFetcher->addParam($dynamicQueryParam);
+
+            $page = $paramFetcher->get('page');
+            $articles = array('bim', 'bam', 'bingo');
+
+            return array('articles' => $articles, 'page' => $page);
+        }
+
+.. note::
+
+    There is also ``$paramFetcher->all()`` to fetch all configured query
+    parameters at once. And also both ``$paramFetcher->get()`` and
+    ``$paramFetcher->all()`` support and optional ``$strict`` parameter to throw
+    a ``\RuntimeException`` on a validation error.
+
+.. note::
+
+    The ParamFetcher requirements feature requires the symfony/validator
+    component.
+
+Optionally the listener can also already set all configured query parameters as
```

Comment:
`[the listener] can also already set all [...]` sounds a bit weird, but I'm not sure if it's wrong.